integration/tests/cook/test_multi_user.py

```diff
@@ -84,3 +84,41 @@ def test_multi_user_usage(self):
             # Terminate all of the jobs
             if all_job_uuids:
                 util.kill_jobs(self.cook_url, all_job_uuids)
+
+    def test_job_too_big(self):
+        admin = self.user_factory.admin()
+        user = self.user_factory.new_user()
+        try:
+            # User with no quota can't submit jobs
+            with admin:
+                resp = util.set_limit(self.cook_url, 'quota', user.name, cpus=0)
+                self.assertEqual(resp.status_code, 201, resp.text)
+            with user:
+                _, resp = util.submit_job(self.cook_url)
+                self.assertEqual(resp.status_code, 422, msg=resp.text)
+            # User with tiny quota can't submit bigger jobs, but can submit tiny jobs
+            with admin:
+                resp = util.set_limit(self.cook_url, 'quota', user.name, cpus=4)
+                self.assertEqual(resp.status_code, 201, resp.text)
+            with user:
+                _, resp = util.submit_job(self.cook_url, cpus=5)
+                self.assertEqual(resp.status_code, 422, msg=resp.text)
+                _, resp = util.submit_job(self.cook_url, cpus=4)
+                self.assertEqual(resp.status_code, 201, msg=resp.text)
```

Comment:
Maybe we should make this case even tinier so that we don't actually use 4 cpus on the cluster when we run the test